bash_completionsV2.go

```diff
@@ -220,7 +220,14 @@ __%[1]s_handle_standard_completion_case() {
 
     # Short circuit to optimize if we don't have descriptions
     if [[ "${completions[*]}" != *$tab* ]]; then
-        IFS=$'\n' read -ra COMPREPLY -d '' < <(compgen -W "${completions[*]}" -- "$cur")
+        compgen_words=$(printf "%%s\n" "${completions[@]}")
+        IFS=$'\n' read -ra COMPREPLY -d '' < <(IFS=$'\n' compgen -W "${compgen_words}" -- "$cur")
+
+        # If there is a single completion left, escape the completion
+        if [ ${#COMPREPLY[*]} -eq 1 ]; then
+            COMPREPLY[0]=$(printf %%q "${COMPREPLY[0]}")
```

Comment:
I think we can eliminate a subshell here.
```suggestion
            printf -v "COMPREPLY[0]" %%q "${COMPREPLY[0]}"
```

Reply:
Hmm, actually `printf -v` to an array index is a bash 4.1+ thing, so as/if older compatibility is needed, then scratch this and similar comments of mine elsewhere.

Reply:
Yeah, we try to keep it working with bash 3, as much as possible.

Reply:
If the printf being executed is not a shell builtin, I don't think `-v foo` as in "assign to variable foo" can work -- the printf executable can not set variables in the parent shell.

I suspect the `which` you were using does not look up from your shell's builtin etc. `type printf` is a more reliable way to check.

```shellsession
$ docker run --rm -it centos:5 bash
[root@388a3ad77194 /]# echo $BASH_VERSION
3.2.25(1)-release
[root@2edc18906647 /]# type printf
printf is a shell builtin
[root@388a3ad77194 /]# printf -v foo foo
[root@388a3ad77194 /]# echo $foo
foo
[root@388a3ad77194 /]# /usr/bin/printf -v bar bar
-v/usr/bin/printf: warning: ignoring excess arguments, starting with `bar'
[root@388a3ad77194 /]# exit
```

Reply:
Anyway `printf -v` is available in bash >= 3.1: https://github.com/bminor/bash/blob/6794b5478f660256a1023712b5fc169196ed0a22/NEWS#L1442-L1443

(Official upstream repo gives 503 at the moment, therefore pointing to a random GH mirror I found.)